### PR TITLE
#168374991 create user account

### PR DESCRIPTION
**Pull request by higustave12**
#### What does this PR do?
create user account endpoint
#### Description of Task to be completed?
create user account endpoint takes (firstName, lastName, email, password, address, bio, occupation and expertise) as request and returns (token, message, firstName, lastName, email, address, bio, occupation and expertise) as response. Also response status code may be 201: for a user account successfully created, 400: for a bad request and 409: for a user trying to create an already existing account.
#### How should this be manually tested?
- Clone this repository to the computer
- Run npm start" into your VS CODE EDITOR console. 
- Fill in the body fields required data
- Open POSTMAN, type "localhost:3000/api/v2/auth/signup" and hit SEND button
#### What are the relevant pivotal tracker stories?
[STORY LINK](https://www.pivotaltracker.com/story/show/168374991)
#### Screenshots (if appropriate)
image
image

### CODE REVIEW

File: SERVER/CONTROLLERS/userAcc_Controler.js [hunk 1]
```diff
@@ -1,71 +1,71 @@
...
-                    occupation: createdAccount.occupation,
-                    expertise: createdAccount.expertise,
-                    isAdmin: createdAccount.isAdmin,
-                    isAmentor: createdAccount.isAmentor
-                };
-
-                const token= jwt.sign(user_token, user_secret, { expiresIn: "1d" });
-                res.header('x-auth-token', token);
+            pool.connect((err, client, done) => {
+                const query = `SELECT * FROM users WHERE email = $1`;
```
Comment (njeri-ngigi) [inline SERVER/CONTROLLERS/userAcc_Controler.js]: Your controller is doing unnecessary work. Its main job should be to pass control to other methods and then return a response.

I would recommend that you create service files that will contain methods that interact with the database. You can place the service files in the database folder or create a new folder under `src` called services

File: SERVER/CONTROLLERS/userAcc_Controler.js [hunk 2]
```diff
@@ -1,71 +1,71 @@
...
+                                const token= jwt.sign(user_token, user_secret, { expiresIn: "1d" });
+                                res.header('x-auth-token', token);
+                                return res.status(201).send({
+                                    status: 201,
+                                    message: `User created successfully`,
+                                    data: {
+                                        token: token,
+                                        id: result.rows[0].userid,
+                                        firstName: result.rows[0].firstname,
+                                        lastName: result.rows[0].lastname,
```
Comment (njeri-ngigi) [inline SERVER/CONTROLLERS/userAcc_Controler.js]: Make use of object destructuring here

Comment (njeri-ngigi) [reply SERVER/CONTROLLERS/userAcc_Controler.js]: Or use the spread operator. Whichever suits you

File: SERVER/CONTROLLERS/userAcc_Controler.js (hunk 1 above)
Comment (higustave12) [reply SERVER/CONTROLLERS/userAcc_Controler.js]: Ok. I am gonna work on it

File: SERVER/CONTROLLERS/userAcc_Controler.js [hunk 3]
```diff
@@ -1,71 +1,59 @@
...
+            pool.connect((err, client, done) => {
+                const values = [email.trim()];
+                client.query(selectQuery, values, (error, result) => {
+                    if (result.rows[0]) {
+                        return res.status(409).send({
+                            status: 409,
+                            error: `This account already exists.Try another one.`
+                        });
+                    }else {
+                        pool.connect((err, client, done) => {
```
Comment (njeri-ngigi) [inline SERVER/CONTROLLERS/userAcc_Controler.js]: I don't understand why you are opening the connection twice. Why not use the same client to check if the user exists & then add the user?

File: SERVER/ROUTES/Routes.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ import {verifyToken} from '../JWT/jwt_conf';
 const router= express.Router();
 
 //User accounts routes
-router.post('/api/v1/auth/signup',account_controler.createUseraccount); //Endpoint to create user account
+router.post('/api/v2/auth/signup',account_controler.createUseraccount); //Endpoint to create user account
```
Comment (njeri-ngigi) [inline SERVER/ROUTES/Routes.js]: you don't need unnecessary comments. Using descriptive variable names is a much better way of documenting your code than commenting.

Use comments for code that is confusing, hard to understand or read, and code that will cause a side effect if changed

File: SERVER/CONTROLLERS/userAcc_Controler.js (hunk 3 above)
Comment (higustave12) [reply SERVER/CONTROLLERS/userAcc_Controler.js]: God morning Shalon, I am going to solve it.

File: SERVER/ROUTES/Routes.js (hunk 4 above)
Comment (higustave12) [reply SERVER/ROUTES/Routes.js]: I am going to remove comments

**Review by njeri-ngigi — APPROVED** [review]

**Pull request merged**